packages/core/src/model.ts

```diff
@@ -66,6 +67,21 @@ export function createModel(
         'context' in config ? config : { ...config, context: initialContext },
         implementations
       );
+    },
+    withInput: (input: any) => {
+      const model = createModel(initialContext, creators as any);
+
+      return {
+        ...model,
+        createMachine: (config, implementations) => {
+          return createMachine(
+            'context' in config
+              ? config
+              : { ...config, context: initialContext },
+            { input, ...implementations }
```

Comment:
q: should this merge with "existing" input or should it overwrite it? I feel like our rules for this sort of thing are somewhat inconsistent, or at least under-specified, and that all our APIs should behave in the same way when it comes to this sort of thing. Unless there is a strong reason why they shouldn't.

At the moment I feel like a newcomer has to consult the docs or the implementation a lot, to know which behavior is used where. And even users who are using XState for a long time already are regularly confused by our APIs not being consistent in such scenarios, a recent tweet: https://twitter.com/rafalfilipek/status/1460236700207271938